src/strings.cpp

```diff
@@ -903,732 +903,736 @@ static void FormatString(StringBuilder &builder, const char *str_arg, StringPara
+				case SCC_VEHICLE_NAME: { // {VEHICLE}
+					const Vehicle *v = Vehicle::GetIfValid(args.GetNextParameter<VehicleID>());
+					if (v == nullptr) break;
+
+					if (!v->name.empty()) {
+						auto tmp_params = MakeParameters(v->name);
+						GetStringWithArgs(builder, STR_JUST_RAW_STRING, tmp_params);
+					} else if (v->group_id != DEFAULT_GROUP) {
+						/* The vehicle has no name, but is member of a group, so print group name */
+						auto tmp_params = MakeParameters(v->group_id, v->unitnumber);
+						GetStringWithArgs(builder, STR_FORMAT_GROUP_VEHICLE_NAME, tmp_params);
+					} else {
+						auto tmp_params = MakeParameters(v->unitnumber);
+
+						StringID string_id;
+						switch (v->type) {
+							default:           string_id = STR_INVALID_VEHICLE; break;
+							case VEH_TRAIN:    string_id = STR_SV_TRAIN_NAME; break;
+							case VEH_ROAD:     string_id = STR_SV_ROAD_VEHICLE_NAME; break;
+							case VEH_SHIP:     string_id = STR_SV_SHIP_NAME; break;
+							case VEH_AIRCRAFT: string_id = STR_SV_AIRCRAFT_NAME; break;
+						}
+
+						GetStringWithArgs(builder, string_id, tmp_params);
+					}
+					break;
 				}
-				break;
-			}
 
-			case SCC_SIGN_NAME: { // {SIGN}
-				const Sign *si = Sign::GetIfValid(args.GetNextParameter<SignID>());
-				if (si == nullptr) break;
+				case SCC_SIGN_NAME: { // {SIGN}
+					const Sign *si = Sign::GetIfValid(args.GetNextParameter<SignID>());
+					if (si == nullptr) break;
 
-				if (!si->name.empty()) {
-					auto tmp_params = MakeParameters(si->name);
-					GetStringWithArgs(builder, STR_JUST_RAW_STRING, tmp_params);
-				} else {
-					auto tmp_params = ArrayStringParameters<0>();
-					GetStringWithArgs(builder, STR_DEFAULT_SIGN_NAME, tmp_params);
+					if (!si->name.empty()) {
+						auto tmp_params = MakeParameters(si->name);
+						GetStringWithArgs(builder, STR_JUST_RAW_STRING, tmp_params);
+					} else {
+						auto tmp_params = ArrayStringParameters<0>();
+						GetStringWithArgs(builder, STR_DEFAULT_SIGN_NAME, tmp_params);
+					}
+					break;
 				}
-				break;
-			}
 
-			case SCC_STATION_FEATURES: { // {STATIONFEATURES}
-				StationGetSpecialString(builder, args.GetNextParameter<StationFacility>());
-				break;
-			}
+				case SCC_STATION_FEATURES: { // {STATIONFEATURES}
+					StationGetSpecialString(builder, args.GetNextParameter<StationFacility>());
+					break;
+				}
 
-			case SCC_COLOUR: { // {COLOUR}
-				StringControlCode scc = (StringControlCode)(SCC_BLUE + args.GetNextParameter<Colours>());
-				if (IsInsideMM(scc, SCC_BLUE, SCC_COLOUR)) builder.Utf8Encode(scc);
-				break;
-			}
+				case SCC_COLOUR: { // {COLOUR}
+					StringControlCode scc = (StringControlCode)(SCC_BLUE + args.GetNextParameter<Colours>());
+					if (IsInsideMM(scc, SCC_BLUE, SCC_COLOUR)) builder.Utf8Encode(scc);
+					break;
+				}
 
-			default:
-				builder.Utf8Encode(b);
-				break;
+				default:
+					builder.Utf8Encode(b);
+					break;
+			}
+		} catch (...) {
```

Comment:
Not sure how feasible an additional debug message might be, with additional information (exception text, at least?)

Reply:
I left the original Debug() calls in, but I guess I can remove them and do a Debug() from here.

But I don't see a way to easily get useful extra information—mostly because we're processing a sub-string, but it's the higher-level string that has the real issue.